tests/integration/vpp/04_acl_test.go

```diff
@@ -0,0 +1,499 @@
+					LowerPort: 1150,
+					UpperPort: 1250,
+				},
+			},
+		},
+	},
+	{
+		//RuleName:  "denyUDP",
+		Action: acl.ACL_Rule_DENY,
+		IpRule: &acl.ACL_Rule_IpRule{
+			Udp: &acl.ACL_Rule_IpRule_Udp{
+				SourcePortRange: &acl.ACL_Rule_IpRule_PortRange{
+					LowerPort: 150,
+					UpperPort: 250,
+				},
+				DestinationPortRange: &acl.ACL_Rule_IpRule_PortRange{
+					LowerPort: 1150,
+					UpperPort: 1250,
+				},
+			},
+		},
+	},
+}
+
+var aclMACIPrules = []*acl.ACL_Rule{
+	{
+		//RuleName:  "denyIPv4",
+		Action: acl.ACL_Rule_DENY,
+		MacipRule: &acl.ACL_Rule_MacIpRule{
+			SourceAddress:        "192.168.0.1",
+			SourceAddressPrefix:  uint32(16),
+			SourceMacAddress:     "11:44:0A:B8:4A:35",
+			SourceMacAddressMask: "ff:ff:ff:ff:00:00",
+		},
+	},
+	{
+		//RuleName:  "denyIPv6",
+		Action: acl.ACL_Rule_DENY,
+		MacipRule: &acl.ACL_Rule_MacIpRule{
+			SourceAddress:        "dead::1",
+			SourceAddressPrefix:  uint32(64),
+			SourceMacAddress:     "11:44:0A:B8:4A:35",
+			SourceMacAddressMask: "ff:ff:ff:ff:00:00",
+		},
+	},
+}
+
+// Test add IP acl rules
+func TestAddIPAcl(t *testing.T) {
+	ctx := setupVPP(t)
+	defer ctx.teardownVPP()
+
+	ih := ifplugin_vppcalls.CompatibleInterfaceVppHandler(ctx.vppBinapi, logrus.NewLogger("test"))
+
+	const ifName = "loop1"
+	ifIdx, errI := ih.AddLoopbackInterface(ifName)
+	if errI != nil {
+		t.Fatalf("creating interface failed: %v", errI)
+	}
+	t.Logf("interface created %v", ifIdx)
+
+	ifIndexes := ifaceidx.NewIfaceIndex(logrus.NewLogger("test-iface1"), "test-iface1")
+	ifIndexes.Put(ifName, &ifaceidx.IfaceMetadata{
+		SwIfIndex: ifIdx,
+	})
+
+	h := aclplugin_vppcalls.CompatibleACLVppHandler(ctx.vppBinapi, ifIndexes, logrus.NewLogger("test"))
+	if h == nil {
+		t.Fatalf("handler was not created")
+	}
+
+	acls, errA := h.DumpACL()
+	if errA != nil {
+		t.Fatalf("dumping acls failed: %v", errA)
+	}
+	t.Logf("%d acls dumped", len(acls))
+	t.Logf("acls dumped %v", acls)
+
+	aclIdx, err := h.AddACL(aclIPrules, "test0")
+	Expect(err).To(BeNil())
+	Expect(aclIdx).To(BeEquivalentTo(0))
+	if err != nil {
+		t.Fatalf("adding acls failed: %v", err)
+	}
+	t.Logf("%d acl added", aclIdx)
+
+	acls, errx := h.DumpACL()
+
+	if errx != nil {
+		t.Fatalf("dumping acls failed: %v", errx)
+	}
+	t.Logf("%d acls dumped", len(acls))
+	t.Logf("acls dumped %v", acls)
+
+	_, err = h.AddACL(aclNoRules, "test1")
+	Expect(err).To(Not(BeNil()))
+	if err != nil {
+		t.Logf("adding acls failed: %v", err)
+	}
+
+	_, err = h.AddACL(aclErr1Rules, "test2")
+	Expect(err).To(Not(BeNil()))
+	if err != nil {
+		t.Logf("adding acls failed: %v", err)
+	}
+
+	_, err = h.AddACL(aclErr2Rules, "test3")
+	Expect(err).To(Not(BeNil()))
+	if err != nil {
+		t.Logf("adding acls failed: %v", err)
+	}
+
+	_, err = h.AddACL(aclErr3Rules, "test4")
+	Expect(err).To(Not(BeNil()))
+	if err != nil {
+		t.Logf("adding acls failed: %v", err)
+	}
+
+	//ctx.MockVpp.MockReply(&acl_api.MacipACLAddReply{})
+	aclIdx, err = h.AddACL(aclIPrules, "test5")
```

Comment:
Why in unit tests at this place is expected to be err not NIL

Reply:
https://github.com/ligato/vpp-agent/blob/dev/plugins/vpp/aclplugin/vppcalls/vpp1908/acl_vppcalls_test.go#L271-L277

This part is testing: invalid reply (wrong reply message) and error reploy (retval != 0). These test cases in this form are not relevant in the integration tests. However it would be good to test cases where error happens as well. For example trying to use invalid configuration and expect error from VPP.. etc.